tests/test_cli.py

```diff
@@ -815,13 +876,67 @@ def test_recover_recovery_staging_path(
 
         # THEN if we expected an error the error was observed
         _, err = capsys.readouterr()
+        errors = [msg for msg in err.split("\n") if msg.startswith("ERROR: ")]
         if should_error:
-            assert len(err) > 0
+            assert len(errors) > 0
+            assert any(
+                [
+                    msg.startswith("ERROR:") and error_substring in msg
+                    for msg in err.split("\n")
+                ]
+            )
         else:
+            assert len(errors) == 0
             # AND if we expected success, the server config recovery staging
             # path matches expectations
             assert server.recovery_staging_path == expected_recovery_staging_path
 
+    @pytest.mark.parametrize(
+        "status",
+        [
+            BackupInfo.DONE,
+            BackupInfo.WAITING_FOR_WALS,
+            BackupInfo.FAILED,
+            BackupInfo.EMPTY,
+            BackupInfo.SYNCING,
+            BackupInfo.STARTED,
+        ],
+    )
+    @patch("barman.output.error")
+    @patch("barman.cli.parse_backup_id")
+    @patch("barman.cli.get_server")
+    def test_recover_recovery_staging_path_backup_status(
+        self,
+        get_server_mock,
+        parse_backup_id_mock,
+        error_mock,
+        status,
+        mock_recover_args,
+    ):
+
+        server = build_mocked_server(name="test_server")
+
+        get_server_mock.return_value = server
+
+        backup_info = build_test_backup_info(
+            server=server,
+            backup_id="test_backup_id",
+            status=status,
+        )
+
+        parse_backup_id_mock.return_value = backup_info
+        mock_recover_args.backup_id = "test_backup_id"
+
+        with pytest.raises(
+            SystemExit,
+        ):
+            recover(mock_recover_args)
+
+            error_mock.assert_called_once_with(
+                "Cannot recover from backup 'test_backup_id' of server "
+                "'test_server': backup status is not DONE"
+            )
+
```

Comment:
```suggestion
```
I don't understand what this test is checking. It seems to me we can remove this.

Reply:
Changed the name of the method to test_recover_backup_status and kept the test!